h/groups/views.py

```diff
@@ -15,13 +15,15 @@
 
 @view_config(route_name='group_create',
              request_method='GET',
-             renderer='h:groups/templates/create.html.jinja2',
-             permission="authenticated")
+             renderer='h:groups/templates/create.html.jinja2')
 def create_form(request):
     """Render the form for creating a new group."""
     if not request.feature('groups'):
         raise exc.HTTPNotFound()
 
+    if request.authenticated_userid is None:
+        raise exc.HTTPNotFound()
```

Comment:
Fair enough. Note, FWIW, that the situation Ron Synder ran into -- which I
just checked myself -- is that 404 is also the response when failing to
create an annotation with bogus permissions. That complicates debugging
auth problems.

On Wed, Sep 16, 2015 at 1:25 PM, Randall Leeds <notifications@github.com>
wrote:

> In h/groups/views.py
> <https://github.com/hypothesis/h/pull/2500#discussion_r39680244>:
>
> >  def create_form(request):
> >      """Render the form for creating a new group."""
> >      if not request.feature('groups'):
> >          raise exc.HTTPNotFound()
> >
> > +    if request.authenticated_userid is None:
> > +        raise exc.HTTPNotFound()
>
> It's a fairly common approach that we've been taking on the HTML
> representations all along. It keeps information about whether something
> exists or not private.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hypothesis/h/pull/2500/files#r39680244>.
>